### DISCUSSION TITLE
Widget's UI problems

### DISCUSSION THREAD

**Discussion by Luciogi** [LinwoodDev/Butterfly#550] [General]
1. Padding
2. Icon size
---
# ~~Add tool Widget~~

<details> <summary>Image</summary>
<p>
- Search bar is too big (I mean padding) [Portrait]
- Label "Add" is collapsing with search bar [Portrait]
- There is extra padding every where, it is very noticeable on android
- Tools under label "Import", they could be in 2 columns [Portrait] 

</p>

<img src="https://github.com/LinwoodDev/Butterfly/assets/84625092/502fcb43-b965-4c71-b1ad-f612ddb40b79">
</details>

---
# ~~Asset Widget~~
<details> <summary>Expand</summary>
- Add button (+) is outside of boarder which is not press able in  [Portrait] mode
- Padding problem as above
### Suggestion
- (+) button might look better along with ▽
<img src="https://github.com/LinwoodDev/Butterfly/assets/84625092/fd2be903-d984-4787-bae9-4366df404301">
</details>

---

# Templates
- Close button should have circle around it (for consistency)
<details> <summary>Image</summary>
<img src=https://github.com/LinwoodDev/Butterfly/assets/84625092/2ea61501-6f9f-42c4-a108-199dc40a0321 >
<img src=https://github.com/LinwoodDev/Butterfly/assets/84625092/0af7e820-fd7f-42cd-9ad4-ba18bba5e6b1 >
</details>

---
# Settings
- Close button should be on right side (for consistency)
- It should have open animation as slide from left to right
- It should be full screen widget like other i.e Files, Components
---
Finding more ...

**Comment by CodeDoctorDE** [comment]
Thanks for reporting. I will look into it. These are problems if you use a smaller font size on android while developing

**Comment by CodeDoctorDE** [comment]
Okay, now I have time to answer.
Firstly the big padding comes from material 3. It should be used to look modern. I don't know how we should fix that: We could break the specs or try to find an alternative way to fix the overlapping.

## The add dialog

* We could put the search box in the next line to avoid these issues.
* On the tool buttons, I wanted to have a quadratic look. Maybe we could try to make the pin button smaller to make the whole button smaller
* With the previous point it should automatically be grouped in two columns.

## Asset Widget

* Thanks I fixed it. Dropdowns currently doesn't support buttons as trailing in material 3.

## Templates

* I just fixed it, thanks for reporting

## Settings

* All close buttons are currently on the left button, I think it's good for the consistency to have it on the left side
* If you are on desktop you will see that it's not a drawer. The components or files dialogs are drawers and doesn't relate to the settings dialog.
* For me personally, I think this dialog looks better than a full page

**Comment by Luciogi** [comment]
>All close buttons are currently on the left button, I think it's good for the consistency to have it on the left side


<details><summary>Images</summary>
<p>

![Screenshot_20231216-113908_Butterfly Nightly](https://github.com/LinwoodDev/Butterfly/assets/84625092/b14817f4-1e7f-4357-b3b2-587ca8c947f4)
![Screenshot_20231216-105741_Butterfly Nightly](https://github.com/LinwoodDev/Butterfly/assets/84625092/2ee02cdb-5325-432f-bc2a-b2c9ac60c965)
![Screenshot_20231216-105749_Butterfly Nightly](https://github.com/LinwoodDev/Butterfly/assets/84625092/f2ae1f62-db89-48d4-94dd-9f844bcdda39)

</p>
</details> 

These three pages have close button on right side, while **Template** and **Settings** have on left side

So all things should have close button on single side
I would suggest to move close button of Template and Settings to right. :)


On desktop having dialog is keep all things in center of screen, But on Android we have small screen devices too, so it looks odd having dialog, making them full screen also resolve issue with template dialog like search bar would have enough space, label "template" and icon next it would be on single line.
Packs dialogue have problem everything is fitted in single vertical space [portrait mode]

**Reply by CodeDoctorDE** [reply]
These three pages are not dialogs, they are drawer pages. But I would put the close button on the left here too.
The advantage here is that you don't need to move the finger too much where you opened the menu.

I will try it out with the full screen dialog. Looks like a good idea to improve the spacing.
>  Packs dialogue have problem everything is fitted in single vertical space [portrait mode]

What do you mean with this?

**Comment by CodeDoctorDE** [comment]
![image](https://github.com/LinwoodDev/Butterfly/assets/20452814/35f3b2ad-5e42-4737-9943-3366827e039d)

What do you think about this style of add dialog. The dialog is now in full screen on mobile and normal on desktop:
![image](https://github.com/LinwoodDev/Butterfly/assets/20452814/5865c0f7-c054-40fe-b60c-71ae8021fb98)

Btw: The icon is not vertically centered because it doesn't look good if you have two columns next to a one with only one column

**Reply by Luciogi** [reply]
Android view is fine

In desktop view I can only see 
- Addition of close button
- The import tools are but on right side
![Screenshot_20231216_135341](https://github.com/LinwoodDev/Butterfly/assets/84625092/cb68b65d-d8ae-4fd7-92a9-1f1bd02b95d4)

>Btw: The icon is not vertically centered because it doesn't look good if you have two columns next to a one with only one column

are you referring to Android view ? right

**Reply by CodeDoctorDE** [reply]
This is the old dialog, you could try it out at https://preview.butterfly.linwood.dev

> are you referring to Android view ? right

I mean the tool buttons in the dialog (the squares).

What do you want to change in the desktop view?

**Reply by Luciogi** [reply]
desktop view is fine too

As for text alignment, My suggestion is to move pin button on top cornor, as shown in new layout
![image](https://github.com/LinwoodDev/Butterfly/assets/84625092/56c3c342-9d15-4dc3-b9a2-68fbb8c277e2)

**Reply by CodeDoctorDE** [reply]
That's a good idea.
I implemented this:
![image](https://github.com/LinwoodDev/Butterfly/assets/20452814/b39e2c00-cadc-4924-86b4-5ea65da1ba92)

![image](https://github.com/LinwoodDev/Butterfly/assets/20452814/2a34e31a-32d2-4923-9f68-8fef7fd62043)

What do you think?

**Reply by Luciogi** [reply]
On Desktop, add some padding 
![image](https://github.com/LinwoodDev/Butterfly/assets/84625092/1a5883c0-0e91-4558-a135-0c4ee37b3613)

**Reply by Luciogi** [reply]
Otherwise it is looking alright

**Reply by CodeDoctorDE** [reply]
Okay thank you very much. I commited the changes, in a few minutes it should be online at https://preview.butterfly.linwood.dev

**Reply by Luciogi** [reply]
Checked android version it looks very nice

**Comment by Luciogi** [comment]
In template dialogue , search bar is touching screen (only on android [portrait mode])

Edit: Drawer's close button got too small

**Reply by CodeDoctorDE** [reply]
I will add a bit padding on the search bar and it should look better
![image](https://github.com/LinwoodDev/Butterfly/assets/20452814/f2b0e598-7bd0-423f-bb2b-a428aeb9eb83)

**Comment by Luciogi** [comment]
- Packs
- Create Pack
- Edit Pack

Make these dialogues  full screen too (android only)

**Reply by CodeDoctorDE** [reply]
The problem is that the full screen dialog has no actions at the bottom. But I don't see a problem with the current design of the dialogs

**Reply by Luciogi** [reply]
From my perspective, use keyboard on mobile device in landscape is too bad, because keyboard takes up whole screen. So user would likely to go to portrait mode, to use Packs dialogue. But in portrait mode , elements are too close to each other.

**Reply by CodeDoctorDE** [reply]
Can you show me what you mean with "elements are too close to each other"?

**Reply by Luciogi** [reply]
![Screenshot_20231216-155124_Butterfly Nightly](https://github.com/LinwoodDev/Butterfly/assets/84625092/d1cd91d4-38ec-4d3d-ac5b-19799c18980c)

**Reply by CodeDoctorDE** [reply]
I see...
I will try to remove some spacing there

**Reply by Luciogi** [reply]
here, these two button doing same function 
![image](https://github.com/LinwoodDev/Butterfly/assets/84625092/c53d9641-3f93-4ddc-9e4c-73ebf460932d)
